include/pybind11/pybind11.h

```diff
@@ -408,6 +417,12 @@ class cpp_function : public function {
                 if (options::show_function_signatures()) signatures += "\n";
             }
         }
+        if (rec->is_operator) {
+            signatures += "\n";
+            signatures += std::to_string(++index) + ". ";
+            signatures += rec->name;
+            signatures += "(self: " + self_type + ", other: object) -> NotImplemented\n";
```

Comment:
Although the type signature of an operator method is actually `(*args: Any, **kwargs: Any) -> NotImplemented` because pybind falls back to returning `NotImplemented` when all other type signatures fail, mypy is specifically looking for the signature implemented here. We could add an additional type signature `(*args: Any, **kwargs: Any) -> NotImplemented` for correctness as well but that feels like overkill to me?